### PR TITLE
[DEV-4733][DEV-4780] Add fields to download

### PR DESCRIPTION
**Pull request by emily-brents**
**Description:**
Adds several fields to downloads based on PO request.
**Technical details:**
Fields added - 
- `prime_award_period_of_performance_start_date`, `prime_award-period_of_perofrmance_current_end_date` to both subcontracts and subgrants, `period_of_performance_potential_end_date` to subcontracts only
- `award_base_action_date` and `award_base_action_date_fiscal_year` to awards level downloads

**Requirements for PR merge:**

1. [X] Unit & integration tests updated
2. [X] API documentation updated
3. [ ] Necessary PR reviewers:
    - [ ] Backend
4. [N/A] Matview impact assessment completed
5. [N/A] Frontend impact assessment completed
6. [X] Data validation completed
7. [N/A] Appropriate Operations ticket(s) created
8. [X] Jira Ticket [DEV-4733](https://federal-spending-transparency.atlassian.net/browse/DEV-4733)& [DEV-4780](https://federal-spending-transparency.atlassian.net/browse/DEV-4780):
    - [ ] Link to this Pull-Request
    - [ ] Performance evaluation of affected (API | Script | Download)
    - [ ] Before / After data comparison

**Area for explaining above N/A when needed:**
```
No change to matviews, no ops tickets needed, no changes to frontent
```

### CODE REVIEW

File: usaspending_api/download/helpers/download_annotation_functions.py [hunk 1]
```diff
@@ -114,6 +119,15 @@ def subaward_annotations():
     annotation_fields = {
         "subaward_action_date_fiscal_year": FiscalYear("subaward__action_date"),
         "prime_award_base_action_date_fiscal_year": FiscalYear("award__date_signed"),
+        "prime_award_period_of_performance_start_date": Cast(
```
Comment (sethstoudenmier) [inline usaspending_api/download/helpers/download_annotation_functions.py:122]: Why do these need to be cast if they are already `DateField` on the Award model?

File: usaspending_api/download/v2/download_column_historical_lookups.py [hunk 2]
```diff
@@ -1207,6 +1211,9 @@
                 ("prime_award_amount", "broker_subaward__award_amount"),
                 ("prime_award_base_action_date", "broker_subaward__action_date"),
                 ("prime_award_base_action_date_fiscal_year", None),  # Annotation is used to create this column
+                ("prime_award_period_of_performance_start_date", None),
```
Comment (sethstoudenmier) [inline usaspending_api/download/v2/download_column_historical_lookups.py:1214]: Can you add the annotation comment to the few lines missing?

File: usaspending_api/download/helpers/download_annotation_functions.py (hunk 1 above)
Comment (emily-brents) [reply usaspending_api/download/helpers/download_annotation_functions.py]: They don't! For some reason i thought the ticket said DateTimeField and not DateField

File: usaspending_api/download/v2/download_column_historical_lookups.py (hunk 2 above)
Comment (emily-brents) [reply usaspending_api/download/v2/download_column_historical_lookups.py]: sure thing

**Pull request merged**